features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseCheckableTextViewMapperTest.kt

```diff
@@ -0,0 +1,289 @@
+
+        whenever(
+            mockTextWireframeMapper.map(
+                eq(mockCheckableTextView),
+                eq(fakeMappingContext),
+                any(),
+                eq(mockInternalLogger)
+            )
+        )
+            .thenReturn(fakeTextWireframes)
+
+        whenever(
+            mockViewBoundsResolver.resolveViewGlobalBounds(
+                mockCheckableTextView,
+                fakeMappingContext.systemInformation.screenDensity
+            )
+        ).thenReturn(fakeViewGlobalBounds)
+
+        whenever(
+            mockColorStringFormatter.formatColorAndAlphaAsHexString(fakeCurrentTextColor, OPAQUE_ALPHA_VALUE)
+        ) doReturn fakeCurrentTextColorString
+
+        testedCheckableTextViewMapper = setupTestedMapper()
+    }
+
+    internal abstract fun setupTestedMapper(): CheckableTextViewMapper<T>
+
+    internal abstract fun mockCheckableTextView(): T
+
+    internal open fun expectedCheckedShapeStyle(checkBoxColor: String): MobileSegment.ShapeStyle? {
+        return if (fakeMappingContext.privacy == SessionReplayPrivacy.ALLOW) {
+            MobileSegment.ShapeStyle(
+                backgroundColor = checkBoxColor,
+                opacity = mockCheckableTextView.alpha
+            )
+        } else {
+            null
+        }
+    }
+
+    // region Unit Tests
+
+    @TestTargetApi(Build.VERSION_CODES.M)
+    @Test
+    fun `M create ImageWireFrame W map() { checked, above M }`() {
+        // Given
+        val allowedMappingContext = fakeMappingContext.copy(privacy = SessionReplayPrivacy.ALLOW)
+        whenever(mockButtonDrawable.intrinsicHeight).thenReturn(fakeIntrinsicDrawableHeight)
+        whenever(mockCheckableTextView.isChecked).thenReturn(true)
+
+        // When
+        testedCheckableTextViewMapper.map(
+            mockCheckableTextView,
+            allowedMappingContext,
+            mockAsyncJobStatusCallback,
+            mockInternalLogger
+        )
+
+        val expectedX = testedCheckableTextViewMapper
+            .resolveCheckableBounds(mockCheckableTextView, fakeMappingContext.systemInformation.screenDensity).x
+        val expectedY = testedCheckableTextViewMapper
+            .resolveCheckableBounds(mockCheckableTextView, fakeMappingContext.systemInformation.screenDensity).y
+        // Then
+        verify(fakeMappingContext.imageWireframeHelper).createImageWireframe(
+            view = eq(mockCheckableTextView),
+            currentWireframeIndex = anyInt(),
+            x = eq(expectedX),
+            y = eq(expectedY),
+            width = eq(mockCloneDrawableIntrinsicWidth),
+            height = eq(mockCloneDrawableIntrinsicHeight),
+            usePIIPlaceholder = anyBoolean(),
+            drawable = eq(mockClonedDrawable),
+            asyncJobStatusCallback = eq(mockAsyncJobStatusCallback),
+            clipping = eq(MobileSegment.WireframeClip()),
+            shapeStyle = eq(null),
+            border = eq(null),
```

Comment:
minor: occurrences of eq(null) can be replaced by isNull()

Reply:
good point, updated